pkg/storegateway/gateway_blocks_http.go

```diff
@@ -156,9 +197,105 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 		ShowDeleted: showDeleted,
 		ShowSources: showSources,
 		ShowParents: showParents,
+		ActionType:  action,
+		InfoText:    infoText,
 	}, blocksPageTemplate, req)
 }
 
+func (s *StoreGateway) BlocksWriteHandler(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	tenantID := vars["tenant"]
+	if tenantID == "" {
+		util.WriteTextResponse(w, "Tenant ID can't be empty")
+		return
+	}
+
+	if err := req.ParseForm(); err != nil {
+		util.WriteTextResponse(w, fmt.Sprintf("Can't parse form: %s", err))
+		return
+	}
+	actionType := req.Form.Get("action_type")
+	if actionType == "" {
+		actionType = string(ActionTypeNone) // Set the default value to "none"
+	}
+
+	if !isValidActionType(ActionType(actionType)) {
+		util.WriteTextResponse(w, fmt.Sprintf("Invalid Action Type: %s\n", actionType))
+		return
+	}
+
+	// Perform the action on selected blocks
+	action := ActionType(actionType)
+
+	blockUlidsString := req.Form.Get("block_ulids")
+	var infoText string
+	if action != ActionTypeNone && blockUlidsString != "" {
```

Comment:
Is it valid to call the action endpoint without an action? What's the point? If it is, I'd at least add a comment here.

Reply:
I would fail with a 400 if someone calls the write endpoint without wanting to write something.